INSTALL.md

```diff
@@ -1,76 +1,217 @@
+```
+
+It `ghc` isn't present, please install it using your operating
+system's package manager. For example, on Debian GNU/Linux, use:
+
+```
+$ sudo apt install ghc
+```
+
+However many operating systems ship an old version of `cabal`; in
+order to use `streamly` you should use `cabal` version 3.0 (or later).
+
+You can check the installed version of `cabal` by using its
+`--version` flag:
+
+```
+$ which cabal
+/usr/bin/cabal
+$ /usr/bin/cabal --version
+cabal-install version 1.24.0.2
+compiled using version 1.24.2.0 of the Cabal library
+```
+
+In this example, the installed `cabal` binary is too old for our
+needs. It would need to be upgraded.
+
+You can ask `cabal` to upgrade itself:
+
+```
+$ cabal install Cabal cabal-install
+Resolving dependencies...
+Downloading base16-bytestring-0.1.1.7...
+... lots of build output ...
+Building cabal-install-3.0.0.0...
+Installed cabal-install-3.0.0.0
+```
+
+`cabal` will usually place its compiled binaries inside the directory
+`$HOME/.cabal/bin`.
+
+You can verify the version of your newly installed binary by using the
+`--version` flag again:
+
+```
+$ $HOME/.cabal/bin/cabal --version
+cabal-install version 3.0.0.0
+compiled using version 3.0.2.0 of the Cabal library
+```
+
+If you have upgraded `cabal`, do be sure to specify `$HOME/.cabal/bin`
+early in your `PATH` shell variable, so that the new binary gets used
+instead of the older version on your system.
+
+```
+$ PATH=$HOME/.cabal/bin:$PATH
+```
+
+Once you have `cabal` version 3.0 (or later) successfully installed,
+please refresh its package list by using `cabal update`.
 
 ```
 $ cabal update
+Downloading the latest package list from hackage.haskell.org
 ```
 
-Create a directory for isolated builds:
+### Prepare Your Build Directory
+
+Next create a build directory, so that you can use `cabal`'s _Isolated
+Builds_ feature.
 
 ```
 $ mkdir streamly-play
+$ cd streamly-play
 ```
 
-Create a `.cabal` file:
+Run `cabal init` to create an initial set of project files:
 
 ```
-$ cd streamly-play
 $ cabal init --minimal --dependency base --dependency streamly
+
+Generating LICENSE...
+... other messages ...
+Generating Main.hs...
+Generating streamly-play.cabal...
+
+Warning: no synopsis given. You should edit the .cabal file and add one.
+You may want to edit the .cabal file and add a Description field.
 ```
 
-You can add any other dependencies as well later on by editing the
-`build-depends` section of the cabal file.
+This invocation sets up a build with two build dependencies: `base`
+and `streamly`.  You can add additional dependencies later, by editing
+the `build-depends` section of the generated cabal file
+`streamly-play.cabal`.  Please see the [Cabal User
+Guide](https://www.haskell.org/cabal/users-guide/) for more
+information on configuring `cabal`.
+
+This invocation also creates a skeletal `Main.hs` program which we
+will use later.
 
-### Using hackage version in repl
+### Try `streamly` at the GHCi REPL
```

Comment:
Shall we say `in the` GHCi REPL instead of `at the`? For example, I usually say `in the shell` rather than `at the shell`.

Reply:
Done.  Looking around, common usage seems to be 'in the REPL' or 'at the REPL prompt'.